libraries/botbuilder-dialogs-adaptive/src/actions/throwException.ts

```diff
@@ -0,0 +1,95 @@
+/**
+ * @module botbuilder-dialogs-adaptive
+ */
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import {
+    BoolExpression,
+    BoolExpressionConverter,
+    ValueExpression,
+    ValueExpressionConverter,
+} from 'adaptive-expressions';
+import {
+    Converter,
+    ConverterFactory,
+    Dialog,
+    DialogConfiguration,
+    DialogContext,
+    DialogEvents,
+    DialogTurnResult,
+} from 'botbuilder-dialogs';
+
+export interface ThrowExceptionConfiguration extends DialogConfiguration {
+    disabled?: boolean | string | BoolExpression;
+    errorValue?: unknown | ValueExpression;
+}
+
+/**
+ * Action which throws an exception declaratively.
+ */
+export class ThrowException<O extends object = {}> extends Dialog<O> implements ThrowExceptionConfiguration {
+    public static $kind = 'Microsoft.ThrowException';
+
+    /**
+     * Initializes a new instance of the [ThrowException](xref:botbuilder-dialogs-adaptive.ThrowException) class.
+     * @param errorValue Optional. Memory property path to use to get the error value to throw.
+     */
+    public constructor(errorValue: unknown) {
```

Comment:
Is the type truly fully unknown here? Or can we use a union of `string | Record<'message', string> | ...`?

Reply:
The `errorValue` property is a `ValueExpression`, however a `ValueExpression` accepts anything as its value. Maybe we can use the input type of `ValueExpression` which is `any | string | Expression`(replace `any` with unknown or something).